tests/sources/helpers/rest_client/test_client.py

```diff
@@ -430,3 +430,33 @@ def test_post_json_body_without_params(self, rest_client) -> None:
         assert len(returned_posts) == DEFAULT_PAGE_SIZE  # only one page is returned
         for i in range(DEFAULT_PAGE_SIZE):
             assert returned_posts[i] == {"id": posts_skip + i, "title": f"Post {posts_skip + i}"}
+
+    def test_request_kwargs(self, mocker) -> None:
+        rest_client = RESTClient(
+            base_url="https://api.example.com",
+            session=Client().session,
+        )
+
+        mocked_send = mocker.Mock()
+        rest_client.session.send = mocked_send  # type: ignore[method-assign]
+
+        rest_client.get(
```

Comment:
@burnash when I call `rest_client.paginate()` here then the test fails because it returns immediately from `paginate()` with `None`. Can you understand why?